### PR TITLE
Add support for s3

### PR DESCRIPTION
**Pull request by lgarvey**
This PR adds support for s3 bucket creation, and granting services access to existing s3 buckets using an IAM role policy.

The schema validation will be fixed in a follow up PR.

The `storage.yml` structure has changed slightly to make it more flexible.  Environments are now listed under an environments key.  This makes it easy to config keys that allow across envs, e.g 

```
mybucket:
    type: s3

    readonly: true   # applies to all environments

    services:
        - web
        - s3proxy

    environments:
        default:
            plan: small

       prod: 
           plan: medium-ha
```

The default key remains under environments.

Further PRs will improve validation/schema checking and add some unit tests.

### CODE REVIEW

**Review by ejayesh — APPROVED** [review]
lgtm

File: copilot-bootstrap.py [hunk 1]
```diff
@@ -438,6 +443,11 @@ def generate_storage(storage_config_file, output, overwrite):
 
     env_config = {}
 
+    click.echo("\n>>> Generating cloudformation\n")
+
+    # Validation TODO: check that the environments list matches what is in the copilot/environments/ dir
```
Comment (yusufsheiqh) [inline copilot-bootstrap.py:448]: Is this part of another ticket?

File: copilot-bootstrap.py [hunk 2]
```diff
@@ -447,38 +457,67 @@ def _lookup_plan(storage_type, env_conf):
...
+
+            click.echo(_mkfile(output, path / f"{storage_name}.yml", contents, overwrite=overwrite))
+
+        # s3 buckets require additional service level cloudformation to grant the ECS task role access to the bucket
+        if storage_type in ["s3", "s3-policy"]:
+
+            template = templates["svc"]["s3-policy"]
+
+            for svc in storage_config.get("services", []):
+                # TODO: should we validate that the svc exists?
```
Comment (yusufsheiqh) [inline copilot-bootstrap.py:506]: Yes.

**Review by yusufsheiqh — APPROVED** [review]

File: copilot-bootstrap.py (hunk 1 above)
Comment (lgarvey) [reply copilot-bootstrap.py:448]: Yep, service and env validation is done in the jsonschema PR

Comment (yusufsheiqh) [reply copilot-bootstrap.py:448]: Resolved in https://github.com/uktrade/copilot-tools/pull/27.

**Pull request merged**